packages/react-aria-components/test/AriaAutocomplete.test-util.tsx

```diff
@@ -0,0 +1,561 @@
+            let {getByRole} = renderers.standard({collectionProps: {onAction}});
+            let input = getByRole('searchbox');
+            let menu = getByRole(collectionNodeRole);
+            expect(input).not.toHaveAttribute('aria-activedescendant');
+
+            await user.tab();
+            expect(document.activeElement).toBe(input);
+
+            await user.keyboard('{ArrowDown}');
+            let options = within(menu).getAllByRole(collectionItemRole);
+            expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+            await user.keyboard('{Enter}');
+            expect(onAction).toHaveBeenCalledTimes(1);
+            expect(onAction).toHaveBeenLastCalledWith('1');
+
+            await user.keyboard('{ArrowDown}');
+            await user.keyboard('{Enter}');
+            expect(onAction).toHaveBeenCalledTimes(2);
+            expect(onAction).toHaveBeenLastCalledWith('2');
+          });
+
+          it('should not trigger the wrapped element\'s onAction when hitting Space', async function () {
+            let {getByRole} = renderers.standard({collectionProps: {onAction}});
+            let input = getByRole('searchbox');
+            let menu = getByRole(collectionNodeRole);
+            expect(input).not.toHaveAttribute('aria-activedescendant');
+
+            await user.tab();
+            expect(document.activeElement).toBe(input);
+
+            await user.keyboard('{ArrowDown}');
+            let options = within(menu).getAllByRole(collectionItemRole);
+            expect(input).toHaveAttribute('aria-activedescendant', options[0].id);
+            await user.keyboard('[Space]');
+            act(() => jest.runAllTimers());
+            expect(onAction).toHaveBeenCalledTimes(0);
+            options = within(menu).queryAllByRole(collectionItemRole);
+            expect(options).toHaveLength(0);
+          });
+
+          it('should update the aria-activedescendant when hovering over an item', async function () {
+            let {getByRole} = renderers.standard({});
+            let input = getByRole('searchbox');
+            let menu = getByRole(collectionNodeRole);
+            let options = within(menu).getAllByRole(collectionItemRole);
+            expect(input).not.toHaveAttribute('aria-activedescendant');
+
+            await user.tab();
+            expect(document.activeElement).toBe(input);
+            // Need to press to set a modality
+            await user.click(input);
+            await user.hover(options[1]);
+            act(() => jest.runAllTimers());
+            expect(input).toHaveAttribute('aria-activedescendant', options[1].id);
+            expect(document.activeElement).toBe(input);
+          });
+        }
+
+        it('should trigger the wrapped element\'s onSelectionChange when hitting Enter', async function () {
+          let {getByRole} = renderers.standard({collectionProps: {onSelectionChange, selectionMode: 'multiple'}});
```

Comment:
instead of passing these props, we should have a multiple selection renderer

I know you're doing it because you repeat subsections of these tests, but i'd prefer to repeat them than to bind us to the RAC API